packages/cli/src/commands/integration/remove.ts

```diff
@@ -0,0 +1,357 @@
+    parsedArguments.flags['--unlink-all']
+  ) {
+    client.output.error(
+      'Cannot specify a project while using the `--unlink-all` flag.'
+    );
+    return 1;
+  }
+
+  const resourceOrIntegrationName = parsedArguments.args[1];
+  const searchForIntegration =
+    parsedArguments.args.length === 2 &&
+    !parsedArguments.flags['--delete'] &&
+    !parsedArguments.flags['--unlink-all'];
+
+  if (searchForIntegration) {
+    return await handleRemoveIntegration(
+      client,
+      team,
+      resourceOrIntegrationName,
+      !!parsedArguments.flags['--yes']
+    );
+  }
+
+  return await handleUnlinkOrDeleteResource(
+    client,
+    team,
+    resourceOrIntegrationName,
+    parsedArguments.args,
+    parsedArguments.flags
+  );
+}
+
+// $ vc integration remove [integration-name]
+async function handleRemoveIntegration(
+  client: Client,
+  team: Team,
+  integrationName: string,
+  skipConfirmation: boolean
+): Promise<number> {
+  client.output.spinner('Retrieving integration…', 500);
+  const integrationConfiguration = await getFirstConfiguration(
+    client,
+    integrationName
+  );
+
+  if (!integrationConfiguration) {
+    client.output.error(`No integration ${chalk.bold(integrationName)} found.`);
+    return 0;
+  }
+
+  // Confirm removal desired
+  if (
+    !skipConfirmation &&
+    !(await confirmIntegrationRemoval(
+      client,
+      integrationConfiguration.slug,
+      team
+    ))
+  ) {
+    client.output.log('Canceled');
+    return 0;
+  }
+
+  // Uninstall Integration
+  try {
+    client.output.spinner('Uninstalling integration…', 1000);
+    await removeIntegration(client, integrationConfiguration, team);
+  } catch (error) {
+    client.output.error(
+      chalk.red(
+        `Failed to remove ${chalk.bold(integrationName)}: ${(error as Error).message}`
+      )
+    );
+    return 1;
+  }
+
+  client.output.success(`${chalk.bold(integrationName)} successfully removed.`);
+  return 0;
+}
+
+async function handleUnlinkOrDeleteResource(
+  client: Client,
+  team: Team,
+  resourceName: string,
+  args: string[],
+  flags: RemoveCommandFlags
+): Promise<number> {
+  client.output.spinner('Retrieving resource…', 500);
+  const resources = await getResources(client, team.id);
+  const targetedResource = resources.find(
+    resource => resource.name === resourceName
+  );
+
+  if (!targetedResource) {
+    client.output.error(`No resource ${chalk.bold(resourceName)} found.`);
+    return 0;
+  }
+
+  if (args.length === 3) {
+    if (flags['--unlink-all']) {
+      client.output.error(
+        'Cannot specify a project when using the `--unlink-all` flag.'
+      );
+      return 1;
+    }
+
+    const unlinkProjectResults = await handleUnlinkProject(
+      client,
+      targetedResource,
+      args[2],
+      !!flags['--yes']
+    );
+    if (unlinkProjectResults !== undefined) {
+      return unlinkProjectResults;
+    }
+  } else if (flags['--unlink-all']) {
```

Comment:
question: do you think of a way we could have an earlier return above and not use the elseif here?